core/font.lua

```diff
@@ -20,19 +23,36 @@ SILE.registerCommand("font", function (options, content)
   if options.features then SILE.settings:set("font.features", options.features) end
   if options.direction then SILE.settings:set("font.direction", options.direction) end
   if options.language then
-    if options.language ~= "und" and icu and icu.canonicalize_language then
-      local newlang = icu.canonicalize_language(options.language)
-      -- if newlang ~= options.language then
-        -- SU.warn("Language '"..options.language.."' not canonical, '"..newlang.."' will be used instead.")
-      -- end
-      options.language = newlang
-    end
+    -- BEGIN OMIKHLEIA HACKLANG
+    -- Commented out. Too soon and or at discrepancy with the "language" command anyway.
+    -- So let document.language is what the user wants, and loadLanguage take care of it
+    --
+    -- if options.language ~= "und" and icu and icu.canonicalize_language then
+    --   local newlang = icu.canonicalize_language(options.language)
+    --   -- if newlang ~= options.language then
+    --     -- SU.warn("Language '"..options.language.."' not canonical, '"..newlang.."' will be used instead.")
+    --   -- end
+    --   options.language = newlang
+    -- end
+    -- END OMIKHLEIA HACKLANG
     SILE.settings:set("document.language", options.language)
-    fluent:set_locale(options.language)
+    -- BEGIN OMIKHLEIA HACKLANG
+    -- Commented out. This is BAD design:
+    -- Fluent maintains a global state (locale) but we could be in a temporary environment
+    -- if the font command has contents, and we'd need this to remain scoped.
+    -- So we should act here!
+    --     fluent:set_locale(options.language)
+    -- END OMIKHLEIA HACKLANG
     SILE.languageSupport.loadLanguage(options.language)
   end
   if options.script then SILE.settings:set("font.script", options.script)
   elseif SILE.settings:get("document.language") then
+    -- BEGIN OMIKHLEIA HACKLANG
+    -- BAD STUFF WARNING: This SILE.languageSupport.languages[] is broken, (nearly always returning nil),
+    -- see comment in languages.lua
+    -- ON THE OTHER HAND, setting font.script doesn't seem to be used in any sensible way...
```

Comment:
The *script* parameter isn't used for all languages or fonts, but some locale and font combinations do make use of it and the shaper then needs this info to correctly shape them. We're passing this on to HarfBuzz if it is set, that's all we need to do with it.